presto-main/src/main/java/com/facebook/presto/operator/DriverAbortSignal.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DriverAbortSignal
+{
+    private final AtomicBoolean aborted = new AtomicBoolean();
```

Comment:
@findepi Here is a micro-benchmark (and its results) for different ways of signaling: https://gist.github.com/haozhun/b1b0ec7a8bbe151ee99f3b803b88ff98

The conclusion is that false-sharing is very harmful. After discussion with @martint and @dain, we decided that it's too painful (hard to maintain) to pad manually. And @martint was against using `@Contended`. The only option it left us was using AtomicBoolean, and bet on the fact that it would be too unlucky if we happen to run into false sharing for an object allocated on heap.

Honestly, we're no expert here. And we might have missed something important in our benchmark. If so, I would be more than glad if you can explain it to me.

When you say "L1 miss", what specifically are you referring to? Are you talking about "invalidated" cache line?